core/src/main/scala/org/apache/spark/security/SocketAuthHelper.scala

```diff
@@ -60,7 +61,7 @@ private[spark] class SocketAuthHelper(val conf: SparkConf) {
           shouldClose = false
         } else {
           writeUtf8("err", s)
-          throw new IllegalArgumentException("Authentication failed.")
+          throw SparkCoreErrors.authenticationFailedError("client")
```

Comment:
Not sure if we should change the exception type here. CC @MaxGekk for more guidance.

Reply:
@cloud-fan, are we allowed to change the exception type here?

Reply:
Also let's not forget to update the `@throws` in the scaladoc, in both `authClient` and `authToServer`.

Reply:
@sandeep-katta ^^

Reply:
It is still `IllegalArgumentException`, don't I need to update scaladoc with `SparkIllegalArgumentException` ?

Reply:
I see your point. I'm okay with keeping the current ones.